src/app.jsx

```diff
@@ -196,59 +253,78 @@ class Application extends React.Component {
         default:
             console.warn('Unhandled event type ', event.type, event.status);
         }
     }
 
-    handleEvent(event) {
+    handleEvent(event, root) {
         switch (event.type) {
         case 'container':
-            this.handleContainerEvent(event);
+            this.handleContainerEvent(event, root);
             break;
         case 'image':
-            this.handleImageEvent(event);
+            this.handleImageEvent(event, root);
             break;
         default:
             console.warn('Unhandled event type ', event.type);
         }
     }
 
-    init() {
-        utils.podmanCall("GetVersion")
+    init(root) {
+        this.checkUserService();
+        utils.podmanCall("GetVersion", {}, root)
                 .then(reply => {
-                    this.setState({ serviceAvailable: true });
-                    this.updateImagesAfterEvent();
-                    this.updateContainersAfterEvent();
+                    this.setState({ [root ? "rootServiceAvailable" : "userServiceAvailable"]: true });
+                    this.updateImagesAfterEvent(root);
+                    this.updateContainersAfterEvent(root);
                     utils.monitor("GetEvents", {},
                                   message => {
-                                      message.parameters && message.parameters.events && this.handleEvent(message.parameters.events);
-                                  },
-                                  () => {
-                                      this.setState({ serviceAvailable: false });
+                                      message.parameters && message.parameters.events && this.handleEvent(message.parameters.events, root);
+                                  }, isRoot => {
+                                      this.setState({ [isRoot ? "rootServiceAvailable" : "userServiceAvailable"]: false
+                                      });
                                   },
+                                  root
                     );
                 })
                 .catch(error => {
-                    if (error.name === "ConnectionClosed")
-                        this.setState({ serviceAvailable: false });
-                    else
+                    if (error.name === "ConnectionClosed") {
+                        this.setState({ [root ? "rootServiceAvailable" : "userServiceAvailable"]: false,
+                                        [root ? "rootContainersLoaded" : "userContainersLoaded"]: true,
+                                        [root ? "rootImagesLoaded" : "userImagesLoaded"]: true
+                        });
+                    } else
                         console.error("Failed to call GetVersion():", error);
                 });
     }
 
     componentDidMount() {
-        this.init();
+        this.init(true);
+        this.init(false);
+    }
+
+    checkUserService() {
+        let argv = ["systemctl", "--user", "status", "io.podman.socket"];
```

Comment:
This now does a lot of systemctl CLI calls, which are relatively expensive and also don't provide monitoring (i. e. send a signal when a service becomes (un)available. Time to copy cockpit's pkg/lib/service.js?

I'm ok with doing a conversion to D-Bus calls in a follow-up, though.

However, `status` is  really a no-go here. That will query the journal, which can take several seconds, and is entirely unnecessary here. Please use `is-active --quiet` instead.

Reply:
> This now does a lot of systemctl CLI calls, which are relatively expensive and also don't provide monitoring

This is called only once when the page is first rendered. Also we don't need monitoring in this step, this just checks if user service exists as well (so at least podman 1.5 or higher) and if it does but is not running (discovered somewhere else) then we suggest starting it.

I can use `is-active` for sure.

Reply:
Or probably `is-enabled`, since `is-active` on non-existing service replies with `inactive`.

Reply:
Hmm, so return code for disabled is the same as for non existing (1). SO I guess the best way is to run it with C locale and check if it contains some string.